### PR TITLE
Upload branch jar again

### PR DESCRIPTION
**Pull request by evansiroky**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [ ] All tests and CI builds passing

### Description

This PR brings back the uploading of a jar with the name of the latest branch. The e2e tests in datatools-ui depend on downloading the latest jar from a particular branch name.

### CODE REVIEW

**Review by landonreed — CHANGES_REQUESTED** [review]
Is https://github.com/ibi-group/datatools-server/issues/321 still a problem? I expect that it probably is.

**Comment by evansiroky** [comment]
It seems to have worked this time: https://travis-ci.org/github/ibi-group/datatools-server/builds/720900545#L11247. Maybe merge this and if the problem happens again, I'll investigate further?

**Review by binh-dam-ibigroup — APPROVED** [review]
I am neutral regarding this and the last CI build went well, so I'm going ahead and pre-approve.

**Comment by landonreed** [comment]
I don't think #321 is fixed. It only will fail on merges to master, but here's the travis snippet that causes the issue:

```
$ ls target/*.jar
target/datatools-server-3.6.1-SNAPSHOT.jar
target/datatools-server-3.7.0.jar
target/dt-v3.2.0-760-g2f3ec00.jar
target/dt-v3.2.0-804-g3a93b0a.jar
$ cp target/dt-*.jar deploy/
$ cp target/dt-*.jar "deploy/dt-latest-$BRANCH_CLEAN.jar"
cp: target ‘deploy/dt-latest-master.jar’ is not a directory
The command "cp target/dt-*.jar "deploy/dt-latest-$BRANCH_CLEAN.jar"" failed and exited with 1 during .
```

The issue is that there are multiple jars that it creates during CI and that it tries to copy all of them to a single filename.

**Comment by evansiroky** [comment]
See proposed fix.

File: .travis.yml [hunk 1]
```diff
@@ -79,12 +79,20 @@ after_success:
...
+- export BRANCH=$(if [ "$TRAVIS_PULL_REQUEST" == "false" ]; then echo $TRAVIS_BRANCH; else echo $TRAVIS_PULL_REQUEST_BRANCH; fi)
+  # Replace forward slashes with underscores in branch name.
+- export BRANCH_CLEAN=${BRANCH//\//_}
   # Create directory that will contain artifacts to deploy to s3.
 - mkdir deploy
   # Display contents of target directory (for logging purposes only).
 - ls target/*.jar
   # Copy packaged jar over to deploy dir.
 - cp target/dt-*.jar deploy/
+  # get first jar file and copy it into a new file that adds the current branch name
```
Comment (landonreed) [inline .travis.yml:92]: Looks good, but I think this comment could be expanded a bit:
```
Get the first jar file and copy it into a new file that adds the current branch name. During a merge to master, there are multiple jar files produced, but they each effectively the same code (there may be slight differences in the version shown in the `pom.xml`, but that's not important for the purposes of creating this "latest branch" jar).
```

**Review by landonreed — APPROVED** [review]
Looks good, but I think the comment needs to say a bit more (see comment).

**Pull request merged**